sklearn/pipeline.py

```diff
@@ -868,6 +869,18 @@ def _validate_transformers(self):
                                 "transform. '%s' (type %s) doesn't" %
                                 (t, type(t)))
 
+    def _validate_transformer_weights(self):
+        if not self.transformer_weights:
+            return
+
+        transformer_names = set(name for name, _ in self.transformer_list)
+        for name in self.transformer_weights:
+            if name not in transformer_names:
+                raise KeyError(
```

Comment:
We typically go with `ValueError` here. In this case the "value" would be the `transformer_weights` dict.

```suggestion
                raise ValueError(
```